sbnobj/Common/Reco/Stub.h

```diff
@@ -6,16 +6,33 @@
 #include "TVector3.h"
 
 namespace sbn {
+  class StubHit {
+    public:
+      float charge; //!< Calibrated and corrected for electron lifetime [#elec]
+      short wire; //!< The wire this hit is on
+      bool ontrack; //!< Whether the hit is also on a track
+  };
+
   class Stub {
   public:
     TVector3 vtx; //!< Interaction Vertex / Start of Stub [cm]
     TVector3 end; //!< End of Stub [cm]
-    float charge; //!< Total charge of stub, corrected for elec. lifetime [#elec]
-    float pitch; //!< Pitch of stub on each wire [cm]
-    geo::PlaneID plane; //!< Plane of stub
-    int nwire; //!< Number of wires/hits contributing charge to stub
 
-    float wire_dist; //!< not used
+    // Per-plane information: sorted so that the "best" plane is the first index
+    std::vector<geo::PlaneID> plane; //!< The plane ID
+
+    std::vector<float> pitch; //!< Pitch of stub on each wire [cm]
+    std::vector<float> trkpitch; //!< Pitch of the matched track on each wire [cm]
+    std::vector<float> vtx_w; //!< Wire coordinate of the vertex on this plane
+    std::vector<short> hit_w; //!< Wire of the end point hit on this plane
+
+    std::vector<std::vector<StubHit>> hits; //!< Hits on each plane. Ordered vtx->end
+
+    /// Helper functions
+    float CoreCharge(unsigned plane_index=0) const; //!< Returns the charge along the core of the stub on the given plane index
+    int CoreNHit(unsigned plane_index=0) const; //!< Returns the number of hits along the core of the stub on the given plane index
```

Comment:
Is there an interface to giving a geo::PlaneID and getting this information? I'm worried this may be a little confusing to have to go through the list of 'planes' which are sorted by quality rather than number. Think this is likely to be misinterpreted...

Reply:
Fixed.